src/supercontrast/provider/handlers/openai_handler.py

```diff
@@ -0,0 +1,260 @@
+        self.generator = SENTIMENT_ANALYSIS_PROMPT | model | parser
+
+    def request(self, request: SentimentAnalysisRequest) -> SentimentAnalysisResponse:
+        result: SentimentAnalysisOutput = self.generator.invoke(
+            {"text": truncate_text(request.text)}
+        )
+        return SentimentAnalysisResponse(score=result.score)
+
+    def get_name(self) -> str:
+        return "OpenAI - Sentiment Analysis"
+
+    @classmethod
+    def init_from_env(cls) -> "OpenAISentimentAnalysis":
+        api_key = os.environ.get("OPENAI_API_KEY")
+        if not api_key:
+            raise ValueError("OPENAI_API_KEY environment variable is not set")
+        return cls()
+
+
+# Task.TRANSLATION
+
+
+class TranslationOutput(pydantic.BaseModel):
+    translation: str = pydantic.Field(description="Translated text")
+
+
+TRANSLATION_PROMPT = PromptTemplate(
+    template="Translate the following text from {src_language} to {target_language}:\n\n{text}\n\n{format_instructions}",
+    input_variables=["src_language", "target_language", "text"],
+    partial_variables={
+        "format_instructions": PydanticOutputParser(
+            pydantic_object=TranslationOutput
+        ).get_format_instructions()
+    },
+)
+
+
+class OpenAITranslate(ProviderHandler):
+    def __init__(self, src_language: str, target_language: str):
+        super().__init__(provider=Provider.OPENAI, task=Task.TRANSLATION)
+        model = ChatOpenAI(
+            **{
+                "temperature": 0,
+                "model_name": OPENAI_MODEL_NAME,
+            }
+        ).bind(response_format={"type": "json_object"})
+        parser = PydanticOutputParser(pydantic_object=TranslationOutput)
+        self.generator = TRANSLATION_PROMPT | model | parser
+        self.src_language = src_language
+        self.target_language = target_language
+
+    def request(self, request: TranslationRequest) -> TranslationResponse:
+        result: TranslationOutput = self.generator.invoke(
+            {
+                "src_language": self.src_language,
+                "target_language": self.target_language,
+                "text": truncate_text(request.text),
+            }
+        )
+        return TranslationResponse(text=result.translation)
+
+    def get_name(self) -> str:
+        return "OpenAI - Translate"
+
+    @classmethod
+    def init_from_env(
+        cls, source_language: str, target_language: str
+    ) -> "OpenAITranslate":
+        api_key = os.environ.get("OPENAI_API_KEY")
+        if not api_key:
+            raise ValueError("OPENAI_API_KEY environment variable is not set")
+        return cls(source_language, target_language)
+
+
+# Task.OCR
+
+
+class OCROutput(pydantic.BaseModel):
+    text: str = pydantic.Field(description="Extracted text from the image")
+    bounding_boxes: list[dict] = pydantic.Field(
+        description="List of bounding boxes for each detected text"
+    )
+
+
+OCR_PROMPT = PromptTemplate(
```

Comment:
same logic as before why include a bounding box in this response?

Reply:
this is broken, I might just remove it from this PR, I was unable to even get the image to send, just wanted to try this at testing

Reply:
Still doesn't work, but removed bounding boxes. I'll fix this in a future PR